firehose.go

```diff
@@ -153,6 +153,16 @@ func (p *Firehose) Send(msg *Message) {
 	}()
 }
 
+// SendSync - blocking send.
+// TODO: we might want a version of this that can be fed a timeout value
+// and listen for cancellation channels. This version is for simplicity.
+func (p *Firehose) SendSync(msg *Message) {
+	msg.SetValue(append(msg.Value(), truncatedRecordTerminator...))
+	p.wg.Add(1)
```

Comment:
Wouldn't the channel block already, while waiting for msg?

Reply:
In theory, yes; my reasoning was this can be called from within a goroutine—whatever might be doing so—while p.wg is otherwise independent of the caller.  In the event p.wg is ever used internal to the library to wait for "all outstanding operations to finish," a dangling/outstanding channel operation might be problematic(?)  I wasn't confident enough to omit it otherwise.

Reply:
Sounds good - better be safe than sorry!